src/okareo/model_under_test.py

```diff
@@ -394,22 +449,49 @@ def run_test(
             model_data: dict = {"model_data": {}}
             if self._has_custom_model():
                 scenario_data_points = self._get_scenario_data_points(scenario_id)
+                datapoint_len = len(scenario_data_points)
 
-                custom_model_invoker = self.models["custom"]["model_invoker"]
-                for scenario_data_point in scenario_data_points:
-                    scenario_input: Union[dict, list, str] = (
-                        scenario_data_point.input_.to_dict()
-                        if isinstance(
-                            scenario_data_point.input_,
-                            ScenarioDataPoinResponseInputType0,
+                if not self._has_custom_batch_model():
```

Comment:
Given that we are integrating more logic into the run_test w/ agent evals, I worry about introducing more logic into the run_test function. We don't have to do this now, but it would be good to have some sort of process function on custom model and custom batch model, that is handed the list of scenario_data_points and goes ahead and processes them, batched or not. Invoke or invoke batch would still be abstract (similar format to our ParallelExecutor class on server)

Reply:
Spoke offline re: this. We will add a ticket to the backlog on Trello for refactoring `run_test` appropriately.